File: solutions/_665_non_decreasing_array/solution.py
from typing import List


class Solution:
    def checkPossibility(self, nums: List[int]) -> bool:
        count = 0
        prev = nums[0]
        for i in range(1, len(nums)):
            if prev > nums[i]:
                count += 1
                if count > 1:
                    return False
                if i >= 2 and nums[i - 2] > nums[i]:
                    continue
            prev = nums[i]

        return True


# https://leetcode.com/problems/non-decreasing-array/

File: solutions/_665_non_decreasing_array/solution.java
package _665_non_decreasing_array;

class Solution {

  public boolean checkPossibility(int[] nums) {
    int count = 0;
    int prev = nums[0];
    for (int i = 1; i < nums.length; i++) {
      if (prev > nums[i]) {
        if (++count > 1) {
          return false;
        }
        if (i - 2 >= 0 && nums[i - 2] > nums[i]) {
          continue;
        }
      }
      prev = nums[i];
    }
    return true;
  }
}
/*
https://leetcode.com/problems/non-decreasing-array/
*/
